### PR TITLE
[APPTS-9430] add events api

### PR DESCRIPTION
**Pull request by jareddlc**
This PR will add support for sending events to platform

### CODE REVIEW

File: test/event.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+const Appc = require('../');
+const helper = require('./lib/helper');
+
+let currentSession;
+let user = global.$config.user;
+
+describe('Appc.Event', function () {
+
+	before(function (done) {
```
Comment (pkclark) [inline test/event.js]: @jareddlc it'd be good to see a test verifying failure when not logged in.

File: test/event.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		Appc.setEnvironment(global.$config.environment);
+
+		Appc.Auth.login(user.username, user.password, function (err, session) {
+			should.not.exist(err);
+			session.should.exist;
```
Comment (pkclark) [inline test/event.js]: @jareddlc per NBB's change, prefer `should.exist(session);` syntax on these tests.

File: lib/event.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+function send(session, name, data, callback) {
+	if (!session || !session.id) {
+		return callback(new Error('session is not valid'));
+	}
+
+	if (!name || typeof name !== 'string') {
+		if (typeof name === 'function') {
+			callback = name;
+		} else if (typeof data === 'function') {
+			callback = data;
```
Comment (pkclark) [inline lib/event.js]: @jareddlc none of this arg shifting on name is needed.  If name is not defined, error out.  However, data and callback should be optional.

Comment (jareddlc) [inline lib/event.js]: i wont be able to guarantee a callback is provided otherwise. However I can throw if its not present.

Comment (pkclark) [inline lib/event.js]: @jareddlc log, don't throw.

Comment (jareddlc) [inline lib/event.js]: so just to clarify, instead of calling back with an error, simply log and return?

Comment (pkclark) [inline lib/event.js]: @jareddlc if there's no callback, yes.

Comment (jareddlc) [inline lib/event.js]: so support both callback and fire and forget?

Comment (pkclark) [inline lib/event.js]: @jareddlc yes.

**Comment by jareddlc** [comment]
pushed changes

File: lib/event.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	}
+
+	if (typeof data !== 'object') {
+		debug('invalid event data');
+		return callback ? callback(new Error('invalid event data')) : new Error('invalid event data');
+	}
+
+	let req = Appc.createRequest(session, '/api/v1/event', 'post', function (err, body, resp) {
+		if (err) {
+			return callback && callback(err, null, resp);
```
Comment (pkclark) [inline lib/event.js]: @jareddlc no logging (in debug or otherwise) here when callback is not defined?

**Comment by jareddlc** [comment]
please review

**Comment by pkclark** [comment]
@jareddlc please add a test case for failure when not logged in (no valid session).

**Comment by pkclark** [comment]
CR good.  Tests good.  Ran FR and worked as expected for session and non-session send calls.  Please include verification steps in PRs containing new features.

**Pull request merged**